### ISSUE TITLE
Throws an exception on ubuntu due to recursive symbolic link to /usr/bin/X11

### ISSUE THREAD

**Issue by gbogard** [WriteThemFirst/approvals-java#74]
On Ubuntu, due to backwards compatibility reasons, there is symbolic link between `/usr/bin/X11` and `/usr/bin`. Programs expect X11 binaries to be in former but Ubuntu places them in the latter. 

This is causing the library to fail on Ubuntu with a `java.nio.file.FileSystemLoopException`.
This causes the library to fail on Ubuntu bases CI servers such as Semaphore CI too. 

I'll publish a fix soon. Meanwhile, here's a complete stack trace : 

```
lambda.infrastructure.code.SSPTemplateEngineSpec *** ABORTED ***
   java.lang.ExceptionInInitializerError:
   at com.github.writethemfirst.approvals.reporters.SupportedOs.<clinit>(SupportedOs.java:45)
   at com.github.writethemfirst.approvals.Reporter.<clinit>(Reporter.java:53)
   at com.github.writethemfirst.approvals.approvers.Approver.<init>(Approver.java:78)
   at lambda.infrastructure.TestUtils$Approbation.$init$(TestUtils.scala:27)
   at lambda.infrastructure.code.SSPTemplateEngineSpec.<init>(SSPTemplateEngineSpec.scala:9)
   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
   at java.lang.Class.newInstance(Class.java:442)
   ...
   Cause: java.io.UncheckedIOException: java.nio.file.FileSystemLoopException: /usr/bin/X11
   at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:88)
   at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:104)
   at java.util.Iterator.forEachRemaining(Iterator.java:115)
   at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
   at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
   at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
   at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
   ...
   Cause: java.nio.file.FileSystemLoopException: /usr/bin/X11
   at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:294)
   at java.nio.file.FileTreeWalker.next(FileTreeWalker.java:372)
   at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:84)
   at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:104)
   at java.util.Iterator.forEachRemaining(Iterator.java:115)
   at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
   at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
   at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
   ...

```

**Comment by tyrcho** [comment]
closed with a6f1126d4952bbed76e05b31e760d82ac87469c4

**Comment by tyrcho** [comment]
@gbogard I have the bug again :(
https://gitlab.com/tyrcho/approvals-demo-scala/-/jobs/271760392
